CHANGELOG.md

```diff
@@ -1,8 +1,297 @@
+    * `App::arg_from_usage` in favor of `App::arg`
+    * `App::args_from_usage` in favor of TODO:
+    * `App::gen_completions` in favor of TODO:
+    * `App::gen_completions_to` in favor of TODO:
+    * `App::settings` in favor of `App::setting(Setting1 | Setting2)`
+    * `App::unset_settings` in favor of `App::unset_setting(Setting1 | Setting2)`
+    * `App::global_settings` in favor of `App::global_setting(Setting1 | Setting2)`
+  * **Arg**
+    * `Arg::empty_values` in favor of TODO:
+  * `ArgMatches::usage` in favor of `App::generate_usage`
+  * **Settings**
+    * `AppSettings::PropagateGlobalValuesDown`
+    * `ArgSettings::Global` in favor of `Arg::global` method
+    * `ArgSettings::Multiple` in favor of `ArgSettings::MultipleValues` and `ArgSettings::MultipleOccurrences`
+  * **Macros**
+    * `arg_enum!` in favor of `ArgEnum` derive macro.
+    * `value_t!` in favor of `ArgMatches::value_of_t`
+    * `value_t_or_exit!` in favor of `ArgMatches::value_of_t_or_exit`
+    * `values_t!` in favor of `ArgMatches::values_of_t`
+    * `values_t_or_exit!` in favor of `ArgMatches::values_of_t_or_exit`
+  * Support for `{n}` in help text
+* **Renamed Methods**
+  * **App**
+    * `App::from_yaml` => `App::from`
+    * `App::arg_from_usage` => `App::arg`
+    * `App::help` => `App::override_help`
+    * `App::usage` => `App::override_usage`
+    * `App::template` => `App::help_template`
+    * `App::get_matches_safe` => `App::try_get_matches`
+    * `App::get_matches_from_safe` => `App::try_get_matches_from`
+    * `App::get_matches_from_safe_borrow` => `App::try_get_matches_from_mut`
+  * **Arg**
+    * `Arg::help` => `Arg::about`
+    * `Arg::from_usage` => `Arg::from`
+    * `Arg::set` => `Arg::setting`
+    * `Arg::unset` => `Arg::unset_setting`
+* **Renamed Variants**
+  * `ArgSettings::CaseInsensitive` => `ArgSettings::IgnoreCase`
+  * `ArgSettings::AllowLeadingHyphen` => `ArgSettings::AllowHyphenValues`
+  * `ArgSettings::EmptyValues` => `ArgSettings::AllowEmptyValues`
+  * `ArgSettings::CaseInsensitive` => `ArgSettings::IgnoreCase`
+* **Renamed Fields**
+  * `Error::message` => `Error::cause`
+* **Changed**
+  * `App::write_help` is now a mutable reference instance method (takes `&mut self`)
+  * `Arg::short` now accepts `char` instead of `&str`
+  * `Arg::validator` now takes first argument as `Fn(&str) -> Result<O, E: ToString>` instead of
+    `Fn(String) -> Result<(), String>`
+  * `Arg::validator_os` now takes first argument as `Fn(&OsStr) -> Result<O, OsString>` instead of
+    `Fn(&OsStr) -> Result<(), OsString>`
+* In usage parser, for options `[name]... --option [val]` results in `ArgSettings::MultipleOccurrences`
+  but `--option [val]...` results in `ArgSettings::MultipleValues` *and* `ArgSettings::MultipleOccurrences`.
+  Before both resulted in the same thing
+* `App` and `Arg` now need only one lifetime
+* Allow empty values is no longer the default
+* UseValueDelimiter is no longer the default
```

Comment:
I wasn't able to confirm this because I am confused on delimiter part of the code. I copied this line from the `v3_changes.md`. Can someone please confirm it?

Reply:
No clue whatsoever. @kbknapp Sorry for tossing the ball around, but maybe you can remember?

Reply:
Can you go ahead with reviewing and merging the PR? We can always update this line when @kbknapp is back with answers.